File: packages/python-sdk/examples/streaming.py
"""Streaming example — watch agent thoughts in real time."""

import asyncio
from commander import CommanderClient


async def main():
    async with CommanderClient(
        api_key="cmd-...",
        base_url="http://localhost:3001",
    ) as client:
        # Execute first to get a session_id
        result = await client.run("analyze this project")
        session_id = result.session_id
        if not session_id:
            print("No session id returned")
            return

        # Stream events from the session
        print("Streaming events...")
        async for event in client.stream(session_id):
            if event.event == "output.delta":
                print(event.data.get("content", ""), end="", flush=True)
            elif event.event == "agent.status":
                print(f"\n[Agent: {event.data.get('status', '')}]")
            elif event.event == "tool_call.started":
                print(f"\n[Tool: {event.data.get('toolName', '?')}]")
            elif event.event == "error.occurred":
                print(f"\n[Error: {event.data.get('error', '')}]")


asyncio.run(main())

File: examples/streaming.ts
#!/usr/bin/env npx tsx
/**
 * Streaming Commander Example
 *
 * Runs a task with real-time SSE streaming to watch the agent think.
 *
 * Usage:
 *   export OPENAI_API_KEY=sk-...
 *   npx tsx examples/streaming.ts
 */
import { executeTask } from '@commander/core';

async function main() {
  const task = 'List 5 best practices for writing secure API endpoints.';

  console.log('Task:', task);
  console.log('---');
  console.log('Agent thoughts will stream below:\n');

  const result = await executeTask({
    task,
    options: {
      stream: true,
      effort: 'simple',
      onStream: (chunk) => {
        if (chunk.type === 'thought') {
          process.stdout.write(`\x1b[2m${chunk.content}\x1b[0m\n`);
        } else if (chunk.type === 'tool_call') {
          process.stdout.write(`\x1b[34m[Tool: ${chunk.toolName}]\x1b[0m\n`);
        } else if (chunk.type === 'output') {
          process.stdout.write(chunk.content);
        }
      },
    },
  });

  console.log('\n---');
  console.log('Status:', result.status);
}

main().catch(console.error);
